pkg/compactor/compactor.go

```diff
@@ -111,6 +111,9 @@ type Config struct {
 	// Allow downstream projects to customise the blocks compactor.
 	BlocksGrouperFactory   BlocksGrouperFactory   `yaml:"-"`
 	BlocksCompactorFactory BlocksCompactorFactory `yaml:"-"`
+
+	// used for unit tests only
+	disableBackgroundValidationInFinishBlockUpload bool `yaml:"-"`
```

Comment:
I think this is a more fitting name since that what's actually gets disabled, not just validation.
```suggestion
	disableCompleteBlockUpload bool `yaml:"-"`
```